Piano/app/src/main/res/layout/fragment_halftonekey.xml

```diff
@@ -3,12 +3,20 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".halftonekey">
-
-    <!-- TODO: Update blank fragment layout -->
-    <TextView
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:text="@string/hello_blank_fragment" />
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    tools:context=".HalfTonePianoKeyFragment"
```

Comment:
you can add a paddingStart here in Framelayout in order to split the half tone keys from each others, so that they placed in the right place in the piano keyboard.